File: java/listview-java/app/src/main/java/com/bmarques/listview/dao/OrderDao.java
package com.bmarques.listview.dao;

import com.bmarques.listview.model.Order;

import java.util.ArrayList;
import java.util.List;

public class OrderDao {

    private final static List<Order> orders = new ArrayList<>();
    private static int idCounter = 1;

    public OrderDao() {
    }

    public void save(Order order) {
        order.setId(idCounter);
        orders.add(order);
        updateIds();
    }

    private void updateIds() {
        idCounter ++;
    }

    public List<Order> findAll() {
        return new ArrayList<>(orders);
    }
}

File: kotlin/ListViewKotlin/app/src/main/java/com/bmarques/listviewkotlin/dao/OrderDao.kt
package com.bmarques.listviewkotlin.dao

import com.bmarques.listviewkotlin.model.Order
import java.util.ArrayList

class OrderDao {

    var orders = ArrayList<Order>()
    var idCounter = 1

    fun save(order: Order) {
        order.updateIds(idCounter)
        orders.add(order)
        updateIds()
    }

    private fun updateIds() {
        idCounter++
    }

    fun findAll(): List<Order> {
        return orders
    }

}
